tests/test_iqm_transpilation.py

```diff
@@ -91,42 +96,49 @@ def test_optimize_single_qubit_gates_reduces_gate_count():
 
 
 def test_optimize_single_qubit_gates_raises_on_invalid_basis():
-    """Test that optimisation pass raises error if gates other than ``RZ`` and ``CZ`` are provided."""
+    """Test that optimization pass raises error if gates other than ``RZ`` and ``CZ`` are provided."""
     circuit = QuantumCircuit(1, 1)
     circuit.h(0)
 
     with pytest.raises(ValueError, match="Invalid operation 'h' found "):
         optimize_single_qubit_gates(circuit)
 
 
-def test_submitted_circuit(adonis_architecture):
+@pytest.mark.skip(reason='generate_initial_layout only works with Star architecture circuits')
+@pytest.mark.parametrize('backend', [IQMFakeAdonis(), IQMFakeDeneb(), IQMFakeAphrodite()])
+def test_optimize_single_qubit_gates_preserves_layout(backend):
```

Comment:
This test is currently always skipped (even for Star-architectures like IQMFakeDeneb), is that intentional?

Reply:
No, I fixed it.